core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2834,26 +2850,35 @@ final class AcquisitionLockTimerTask extends TimerTask {
         private final String memberId;
         private final long firstOffset;
         private final long lastOffset;
+        private boolean hasExpired;
 
         AcquisitionLockTimerTask(long delayMs, String memberId, long firstOffset, long lastOffset) {
             super(delayMs);
             this.expirationMs = time.hiResClockMs() + delayMs;
             this.memberId = memberId;
             this.firstOffset = firstOffset;
             this.lastOffset = lastOffset;
+            this.hasExpired = false;
         }
 
         long expirationMs() {
             return expirationMs;
         }
 
+        boolean hasExpired() {
+            return hasExpired;
```

Comment:
This seems to me to be accessed on multiple threads. It is checked underneath the share-partition write lock to determine whether to run the task following a write error, but it can also change if the timer task runs normally.